app/static/favorite_quotes.js

```diff
@@ -0,0 +1,49 @@
+// Adding event listeners
+let hearts = Array.from(document.getElementsByClassName("heart"));
+window.addEventListener("load", function () {
+  hearts.forEach((heart) => {
+    if (heart) {
+      heart.addEventListener("click", on_heart_click);
+    }
+  });
+});
+
+/**
+ * @summary This function is a handler for the event of heart-click.
+ * Whenever a user clicks on a heart icon, in case of empty heart:
+ * saves quote in favorites, as well as changing
+ * the heart icon from empty to full.
+ * In case of full heart:
+ * Removes it and switch back to empty heart icon.
+ * Uses the save_or_remove_quote function to handle db operations.
+ */
+function on_heart_click() {
+  const quote = this.parentNode.previousElementSibling.innerText;
+  const author_element = this.parentNode;
+
+  if (author_element.innerHTML.includes("\\")) {
+    const author = author_element.innerHTML.split("\\ ")[1].split("\n")[0];
+  }
```

Comment:
Use data attributes instead. Create `data-author` and use it to store the author name

Reply:
Turns out the author data is irrelevant to the favorite quotes feature, as opposed to the quotes feature itself, which is already done. So I'm removing this part. However, I'll use this method for your next comment. Thanks!